gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -615,15 +615,33 @@ case class TransformPostOverrides(isAdaptiveContext: Boolean) extends Rule[Spark
   def transformColumnarToRowExec(plan: ColumnarToRowExec): SparkPlan = {
     if (columnarConf.enableNativeColumnarToRow) {
       val child = replaceWithTransformerPlan(plan.child)
-      logDebug(s"ColumnarPostOverrides ColumnarToRowExecBase(${child.nodeName})")
-      val nativeConversion =
-        BackendsApiManager.getSparkPlanExecApiInstance.genColumnarToRowExec(child)
-      val validationResult = nativeConversion.doValidate()
-      if (validationResult.isValid) {
-        nativeConversion
-      } else {
-        TransformHints.tagNotTransformable(plan, validationResult)
+      def isChildGlutenPlan(p: SparkPlan): Boolean = {
```

Comment:
Can we pull out `isGlutenPlan` and `isVanillaColumnarPlan` ?

Reply:
Sure.

Reply:
Updated, pull out plan related util methods to a separate file.